src/OpenLoco/Windows/VehicleList.cpp

```diff
@@ -27,36 +51,853 @@ namespace OpenLoco::Ui::Windows::VehicleList
+
+        updateVehicleList(self);
+        updateVehicleList(self);
+        updateVehicleList(self);
+
+        self->invalidate();
+    }
+
+    // 0x004C2640
+    static void event_08(window* self)
+    {
+        self->flags |= WindowFlags::not_scroll_view;
+    }
+
+    // 0x004C2648
+    static void event_09(window* self)
+    {
+        if (self->flags & WindowFlags::not_scroll_view)
         {
-            registers regs;
-            regs.eax = (uint32_t)companyId;
-            regs.ebx = type;
-            call(0x004C1A05, regs);
-            vehicle_list = (window*)regs.esi;
+            self->row_hover = -1;
         }
+    }
+
+    // 0x004C265B
+    static void getScrollSize(window* self, uint32_t scrollIndex, uint16_t* scrollWidth, uint16_t* scrollHeight)
+    {
+        *scrollHeight = self->var_83C * self->row_height;
+    }
+
+    // 0x004C266D
+    static cursor_id cursor(window* self, int16_t widgetIdx, int16_t xPos, int16_t yPos, cursor_id fallback)
+    {
+        if (widgetIdx != Widx::scrollview)
+            return fallback;
+
+        uint16_t currentIndex = yPos / self->row_height;
+        if (currentIndex < self->var_83C && self->row_info[currentIndex] != -1)
+            return cursor_id::hand_pointer;
+
+        return fallback;
+    }
+
+    // 0x004C26A4
+    static void onScrollMouseOver(window* self, int16_t x, int16_t y, uint8_t scroll_index)
+    {
+        Input::setTooltipTimeout(2000);
+
+        self->flags &= ~WindowFlags::not_scroll_view;
+
+        uint16_t currentRow = y / self->row_height;
+        if (currentRow <= self->var_83C)
+            self->row_hover = self->row_info[currentRow];
         else
+            self->row_hover = -1;
+
+        string_id tooltipId = StringIds::buffer_337;
+        if (self->row_hover == -1)
+            tooltipId = StringIds::null;
+
+        char* tooltipBuffer = const_cast<char*>(StringManager::getString(StringIds::buffer_337));
+
+        // Have we already got the right tooltip?
+        if (tooltipBuffer[0] != '\0' && self->widgets[Widx::scrollview].tooltip == tooltipId && self->row_hover == self->var_85C)
+            return;
+
+        self->widgets[Widx::scrollview].tooltip = tooltipId;
+        self->var_85C = self->row_hover;
+        Ui::ToolTip::closeAndReset();
+
+        if (self->row_hover == -1)
+            return;
+
+        // Initialise tooltip buffer.
+        char* buffer = StringManager::formatString(tooltipBuffer, StringIds::vehicle_list_tooltip_load);
+
+        // Append load to buffer.
+        auto head = EntityManager::get<VehicleHead>(self->var_85C);
+        buffer = head->generateCargoTotalString(buffer);
+
+        printf("%s\n", tooltipBuffer);
+
+        // Figure out what stations the vehicle stops at.
+        auto orders = Vehicles::OrderRingView(head->orderTableOffset);
+        bool isFirstStop = true;
+        for (auto order : orders)
```

Comment:
i need to really make this harder to mess up but you MUST take a reference `auto& order`

Reply:
Thanks, that does indeed fix it. I wonder if we can enforce using the value as a reference at all times?

Reply:
Simplest way would be to return a pointer instead of a reference